redhat-appstudio/service-provider-integration-operator
# feat: adding action json key to audit logs based on ADR 6

### What does this PR do?
Adds `action` json key to audit logs where it makes sense.

Note that there are no specific cases mentioned in the ADR 6 just the following:
![image](https://user-images.githubusercontent.com/73115616/226915076-91e5d068-419d-46f0-a346-76f9a5bcff67.png)
To read more: https://github.com/redhat-appstudio/book/blob/main/ADR/0006-log-conventions.md
So it depends on the preference for which logs should contain which action.

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://issues.redhat.com/browse/SVPI-412

### How to test this PR?

1. deploy
2. `kubectl apply -f samples/spiaccesscheck.yaml`
3. Look at operator logs, they should contain log entry containing "action": "UPDATE" pair similar to this:
![image](https://user-images.githubusercontent.com/73115616/226927541-a717058b-fb3c-494a-aa0d-7a33bcdc2f1d.png)


```
quay.io/redhat-appstudio/pull-request-builds:spi-controller-#
quay.io/redhat-appstudio/pull-request-builds:spi-oauth-#
```
